### PR TITLE
fix: target to json without default values

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

[unnecessary attributes to the target](https://github.com/sasjs/cli/issues/552)

## Intent

Export function to have existing values only

## Implementation

Updated `target.ts`
- Added new function `toJsonNoDefaults`

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] All unit tests are passing (`npm test`).
- [x] All `sasjs-cli` unit tests are passing (`npm test`).

mocked:
image

- [x] Reviewer is assigned.

### CODE REVIEW

File: src/types/target.ts [hunk 1]
```diff
@@ -254,4 +254,32 @@ export class Target implements TargetJson {
...
+      allowInsecureRequests: this.allowInsecureRequests,
+      appLoc: this.appLoc,
+      macroFolders: this.macroFolders,
+      programFolders: this.programFolders,
+      docConfig: this.docConfig,
+      authConfig: this.authConfig,
+      buildConfig: this.buildConfig,
+      jobConfig: this.jobConfig,
+      serviceConfig: this.serviceConfig,
+      streamConfig: this.streamConfig,
```
Comment (krishna-acondy) [inline src/types/target.ts:272]: Won't this cause `undefined` to be returned when a property has not been set?

Comment (saadjutt01) [reply src/types/target.ts:272]: if a property is undefined in an object, means it doesn't exist.

And that's exactly we want, when updating a target, it shouldn't add additional attributes to current.

Comment (krishna-acondy) [reply src/types/target.ts:272]: Yeah, but if for example `this.streamConfig` is undefined, then this function will return an object containing `streamConfig: undefined`. Is that what we want?

Comment (krishna-acondy) [reply src/types/target.ts:272]: Also, do we really need this to be a separate function, or can we just modify the original one to not return the defaults?

Comment (saadjutt01) [reply src/types/target.ts]: Removed new function, added checks in current `toJson` and passed boolean `withDefaults`.

**Review by krishna-acondy — APPROVED** [review]

**Pull request merged**